src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1135,6 +1136,53 @@ private void RunConnectRequestHandlerTask(ConnectParams connectParams)
             }).ContinueWithOnFaulted(null);
         }
 
+        /// <summary>
+        /// Handle new change password requests
+        /// </summary>
+        /// <param name="connectParams"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        protected async Task HandleChangePasswordRequest(
+            ChangePasswordParams changePasswordParams,
+            RequestContext<PasswordChangeResponse> requestContext)
+        {
+            Logger.Write(TraceEventType.Verbose, "HandleChangePasswordRequest");
+            PasswordChangeResponse newResponse = new PasswordChangeResponse();
+            try
+            {
+                RunChangePasswordRequestHandlerTask(changePasswordParams);
+                newResponse.result = true;
+            }
+            catch (Exception ex)
+            {
+                newResponse.result = false;
+                newResponse.ErrorMessage = ex.Message;
+                newResponse.Messages = ex.InnerException != null ? ex.InnerException.Message : ex.Message;
+            }
+            await requestContext.SendResult(newResponse);
+        }
+
+        public void RunChangePasswordRequestHandlerTask(ChangePasswordParams changePasswordParams)
+        {
+            // Empty passwords are not valid.
+            Boolean isPasswordEmpty = string.IsNullOrEmpty(changePasswordParams.NewPassword);
+            if (isPasswordEmpty)
+            {
+                throw new Exception(SR.ConnectionServiceEmptyPassword);
+            }
+
+            // result is null if the ConnectParams was successfully validated
+            ConnectionCompleteParams result = ValidateConnectParams(changePasswordParams);

```

Comment:
returning `null` on success is an unusual pattern in C#.  It looks like this is reusing an existing function so probably fine to leave as-is.